rotkehlchen/exchanges/htx.py

```diff
@@ -298,17 +305,19 @@ def _query_deposits_withdrawals(
 
             try:
                 coin = asset_from_htx(movement['currency'])
-                movements.append(AssetMovement(
-                    timestamp=timestamp,
+                movements.extend(create_asset_movement_with_fee(
+                    timestamp=ts_sec_to_ms(timestamp),
                     location=Location.HTX,
-                    category=category,
-                    address=movement['address'],
-                    transaction_id=movement['tx-hash'],
+                    event_type=event_type,
                     asset=coin,
                     amount=deserialize_asset_amount(movement['amount']),
                     fee_asset=coin,
                     fee=deserialize_fee(movement.get('fee', '0')),
-                    link=str(movement['id']),
+                    unique_id=str(movement['id']),
+                    extra_data=maybe_set_transaction_extra_data(
+                        address=movement['address'],
+                        transaction_id=movement['tx-hash'],
```

Comment:
why not use `get_key_if_has_val()` here?

Reply:
I assumed that since it wasn't being used here before that these were guaranteed to be present, but I checked the [htx docs](https://www.htx.com/en-us/opend/newApiPages/?id=7ec4f050-7773-11ed-9966-0242ac110003) and it does say that at least `tx-hash` may not always be present, so I've changed these to use `get_key_if_has_val()`.